File: exe7.go
package main

import "fmt"

type Professor struct {
    Nome string
}

type Escola struct {
    Nome      string
    Professores []Professor
}

func (e *Escola) AdicionarProfessor(professor Professor) {
    e.Professores = append(e.Professores, professor)
}

func (e Escola) ListarProfessores() {
    fmt.Printf("Professores da escola %s: ", e.Nome)
    for _, professor := range e.Professores {
        fmt.Printf("%s ", professor.Nome)
    }
    fmt.Println()
}

func main() {
    escola1 := Escola{Nome: "Escola A"}
    escola2 := Escola{Nome: "Escola B"}

    professor1 := Professor{Nome: "Professor João"}
    professor2 := Professor{Nome: "Professor Maria"}

    escola1.AdicionarProfessor(professor1)
    escola1.AdicionarProfessor(professor2)
    escola2.AdicionarProfessor(professor1) 

    escola1.ListarProfessores()
    escola2.ListarProfessores()
}

File: exe7.py
class Professor:
    def __init__(self, nome):
        self.nome = nome

    def __str__(self):
        return self.nome

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        self.professores.append(professor)

    def listar_professores(self):
        print(f"Professores da escola {self.nome}: {', '.join(str(p) for p in self.professores)}")

if __name__ == "__main__":
    escola1 = Escola("Escola A")
    escola2 = Escola("Escola B")

    professor1 = Professor("Professor João")
    professor2 = Professor("Professor Maria")

    escola1.adicionar_professor(professor1)
    escola1.adicionar_professor(professor2)
    escola2.adicionar_professor(professor1)  

    escola1.listar_professores()
    escola2.listar_professores()
